x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Analyzer.java

```diff
@@ -1216,7 +1217,33 @@ private Expression implicitCast(Expression e) {
         }
     }
 
-    public static class PruneSubqueryAliases extends AnalyzerRule<SubQueryAlias> {
+    public static class ReplaceSubQueryAliases extends AnalyzerRule<UnaryPlan> {
+
+        @Override
+        protected LogicalPlan rule(UnaryPlan plan) {
+            if (plan.child() instanceof SubQueryAlias) {
+                SubQueryAlias a = (SubQueryAlias) plan.child();
+                return plan.transformExpressionsDown(FieldAttribute.class, f -> {
+                   if (f.qualifier() != null && f.qualifier().equals(a.alias())) {
+                       // Find the underlying concrete relation (EsIndex) and its name as the new qualifier
+                       List<LogicalPlan> children = a.collectFirstChildren(p -> p instanceof EsRelation);
```

Comment:
Wondering if you have any better ideas for getting only the first child, and not first children out of which to extract the first. Not a biggie, though.

Reply:
Those methods are generic since they reside on the `Node` superclass. Creating a `collectFirstChild` method on `UnaryPlan` that just returns `children.get(0)` doesn't make much of an improvement imho.